File: coffee_tea_app/frontend/src/api/menu.ts
import api from './client'

export interface MenuItem {
  item_id: string
  name: string
  category: string
  description: string | null
  price: number
  image_url: string | null
  is_available: boolean
  config_json: Record<string, string[]> | null
  tags: string[] | null
}

export const getMenuItems = (category?: string) =>
  api.get<MenuItem[]>('/menu/', { params: category ? { category } : {} }).then((r) => r.data)

export const getCategories = () =>
  api.get<{ categories: string[] }>('/menu/categories').then((r) => r.data.categories)

File: backend/app/schemas/menu.py
from pydantic import BaseModel
from typing import Optional, Dict, List, Any


class MenuItemCreate(BaseModel):
    item_id: Optional[str] = None
    name: str
    category: str
    description: Optional[str] = None
    price: float
    image_url: Optional[str] = None
    is_available: bool = True
    config_json: Optional[Dict[str, List[str]]] = None
    tags: Optional[List[str]] = None


class MenuItemUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    price: Optional[float] = None
    image_url: Optional[str] = None
    is_available: Optional[bool] = None
    config_json: Optional[Dict[str, List[str]]] = None
    tags: Optional[List[str]] = None


class MenuItem(BaseModel):
    item_id: str
    name: str
    category: str
    description: Optional[str]
    price: float
    image_url: Optional[str]
    is_available: bool
    config_json: Optional[Dict[str, List[str]]]
    tags: Optional[List[str]]
